### PR TITLE
fix: Adds liveness sidecar to v1.14+ helm charts

### PR DESCRIPTION
**Pull request by wilmardo**
# Describe what this PR does #

Follow-up of #438 

Helm charts where split in the time the PR was open so I hasn't been added to the new helm charts.

### CODE REVIEW

File: deploy/cephfs/kubernetes/v1.14+/helm/templates/nodeplugin-daemonset.yaml [hunk 1]
```diff
@@ -108,6 +108,28 @@ spec:
             - name: keys-tmp-dir
               mountPath: /tmp/csi/keys
           resources:
+{{ toYaml .Values.nodeplugin.plugin.resources | indent 12 }}
```
Comment (Madhu-1) [inline deploy/cephfs/kubernetes/v1.14+/helm/templates/nodeplugin-daemonset.yaml]: can you update provisioner helm chat for the same? for both rbd and cephfs

Comment (wilmardo) [reply deploy/cephfs/kubernetes/v1.14+/helm/templates/nodeplugin-daemonset.yaml]: Good catch! Missed it being added there :)

**Comment by humblec** [comment]
@wilmardo please revisit :)

**Comment by wilmardo** [comment]
@humblec Rebase I guess? Will do in a couple minutes :)

**Comment by wilmardo** [comment]
@humblec Rebased against latest master

File: deploy/rbd/kubernetes/v1.14+/helm/ceph-csi-rbd/templates/nodeplugin-daemonset.yaml [hunk 2]
```diff
@@ -107,6 +107,28 @@ spec:
             - name: keys-tmp-dir
               mountPath: /tmp/csi/keys
           resources:
+{{ toYaml .Values.nodeplugin.plugin.resources | indent 12 }}
+        - name: liveness-prometheus
+          image: "{{ .Values.nodeplugin.plugin.image.repository }}:{{ .Values.nodeplugin.plugin.image.tag }}"
+          args:
+            - "--type=liveness"
+            - "--endpoint=$(CSI_ENDPOINT)"
+            - "--livenessport=8081"
```
Comment (humblec) [inline deploy/rbd/kubernetes/v1.14+/helm/ceph-csi-rbd/templates/nodeplugin-daemonset.yaml]: @Madhu-1  What about making this `port` a configurable value . May be in another PR?

Comment (Madhu-1) [reply deploy/rbd/kubernetes/v1.14+/helm/ceph-csi-rbd/templates/nodeplugin-daemonset.yaml]: yes, this can be added to Value.yaml @wilmardo isn't it?

 also, we can add all ceph-csi and sidecar configuration parameters to value.yaml  @wilmardo sounds like a good idea?

Comment (wilmardo) [reply deploy/rbd/kubernetes/v1.14+/helm/ceph-csi-rbd/templates/nodeplugin-daemonset.yaml]: Yeah had this as an improvement but decided to keep this PR simple, just to sync 1.14+ with the 1.13 charts.

Will follow up this PR with those values split into the values.yaml of all the charts :)

Comment (Madhu-1) [reply deploy/rbd/kubernetes/v1.14+/helm/ceph-csi-rbd/templates/nodeplugin-daemonset.yaml]: @wilmardo Thanks :)

**Comment by kfox1111** [comment]
Question. whats the point of 1.14 vs 1.13 charts? helm can template out things differently if need be for different versions.

**Comment by kfox1111** [comment]
The changes look minimal but related to having snapshotting enabled. You can use .Capabilities.KubeVersion.Minor in the helm chart to know what version you are running in. So could wrap all the snapshotting in:
{{- if ge .Capabilities.KubeVersion.Minor 14 }}
Snapshot specific things.
{{- end }}
And then get back to only needing one chart.

**Comment by wilmardo** [comment]
@kfox1111 Thanks! I will look into this and if it is indeed so simple I might join the helm charts!
Thanks for all the pointers 👍
